integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.spark.thriftserver
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.hive.thriftserver.HiveThriftServer2
+
+import org.apache.carbondata.common.logging.LogServiceFactory
+import org.apache.carbondata.core.util.CarbonProperties
+
+object CarbonThriftServer {
+
+  def main(args: Array[String]): Unit = {
+
+    import org.apache.spark.sql.CarbonSession._
+
+    val sparkConf = new SparkConf(loadDefaults = true)
+    val builder = SparkSession
+      .builder()
+      .config(sparkConf)
+      .appName("Carbon Thrift Server")
+      .enableHiveSupport()
+
+    val sparkHome = System.getenv.get("SPARK_HOME")
+    if (null != sparkHome) {
+      builder.config("carbon.properties.filepath",
```

Comment:
In this case, carbon property must be put in SPARK_HOME/conf folder, can user give the path by  spark.driver.extraJavaOptions in spark-default.conf?